Src/FluentAssertions/Types/PropertyInfoSelector.cs

```diff
@@ -65,7 +66,18 @@ public PropertyInfoSelector ThatAreDecoratedWith<TAttribute>()
         public PropertyInfoSelector ThatAreDecoratedWithOrInherit<TAttribute>()
             where TAttribute : Attribute
         {
-            selectedProperties = selectedProperties.Where(property => CustomAttributeExtensions.GetCustomAttributes(property, true).OfType<TAttribute>().Any());
+            // TODO: There seems to be a bug with `GetCustomAttributes` when used as extension method
```

Comment:
It seems that there is a bug with `GetCustomAttributes`, I'll file an issue on Microsoft side.